### ISSUE TITLE
"CORD-19 Reference Title" dataset and MEDLINE/PubMed models

### ISSUE THREAD

**Issue by usuyama** [tensorflow/hub#695]
Thanks for releasing the PubMed models and benchmark results.

We're interested in learning more about the **CORD-19 Reference Title** dataset.
Could you provide us with more details e.g. dataset size, train/val/test split, etc.? 

Will you release the fine-tuning scripts for these datasets?

Task | Description
-- | --
**CORD-19 Reference Title** | **The CORD-19 Reference Title task is a task derived from the CORD-19 dataset[4] which contains scholarly articles about COVID-19. This task is a sentence pair classification task where the first sentence is a sentence from the article containing a reference (e.g., "Roses are red [1]."). The second sentence is a title from the article references (e.g., "A Survey of Roses"). The label is 1 if the sentence and title match or 0 if the title was randomly selected (from another reference in that article).**
CoLA | The Corpus of Linguistic Acceptability (CoLA)[5] consists of English acceptability judgments drawn from books and journal articles on linguistic theory. Each example is a sequence of words annotated with whether it is a grammatical English sentence.
MRPC | The Microsoft Research Paraphrase Corpus (MRPC)[6] is a corpus of sentence pairs automatically extracted from online news sources, with human annotations for whether the sentences in the pair are semantically equivalent.

https://tfhub.dev/google/collections/experts/bert/1
https://tfhub.dev/google/experts/bert/pubmed/1

**Comment by kempy** [comment]
The task is generated from the 2020-04-24 scrape of the CORD-19 dataset as described above. Here are some more details that may be helpful. Our eval dataset includes splits for train (63350 examples), test (8041 examples), and validation (7773 examples). Sentences are sampled with a minimum of 10 tokens and maximum of 1024 tokens. Examples are produced with ~equal numbers of positive and negative examples - a reference title and a sentence which cites that reference are produced, and whenever possible, the same reference title and a sentence that does not cite that reference is also produced.

At this time, we have no plans to publish the generated dataset.

The model can be fine-tuned with the https://github.com/tensorflow/models/blob/master/official/nlp/bert/run_classifier.py script from the TF Model Garden using max_seq_length=128 and --hub_module_url with the relevant TFHub.dev model handle.

**Comment by usuyama** [comment]
Thank you, @kempy @rmothukuru @j-grimstad 

Is this PubMed model pretrained from scratch (pretraining from random weights)? Or is it continual pretraining from the regular bert-base-uncased?

The model ( https://tfhub.dev/google/experts/bert/pubmed/1 ) uses the regular vocab, so I'm guessing it's continual pretraining, but `500K steps with a batch size of 2048` sounds too much for continual pretraining.

**Comment by kempy** [comment]
Oops, sorry for the late reply @usuyama. It is trained from scratch on the PubMed corpus, but uses the original BERT vocab since there was no significant difference in our experiments from using a custom vocab.

**Comment by arghyaganguly** [comment]
@usuyama,please reopen this if you have more questions in future.Thanks.
